ament_cmake_target_dependencies/cmake/ament_target_dependencies.cmake

```diff
@@ -33,10 +33,21 @@ function(ament_target_dependencies target)
     message(FATAL_ERROR "ament_target_dependencies() the first argument must be a valid target name")
   endif()
   if(${ARGC} GREATER 0)
+    set(options PUBLIC)
+    set(oneValueArgs)
+    set(multiValueArgs)
+    cmake_parse_arguments(AMENT_TARGET_DEPENDENCIES
+      "${options}"
+      "${oneValueArgs}"
+      "${multiValueArgs}"
+      ${ARGN})
```

Comment:
This implies that the `PUBLIC` keyword can be passed anywhere in the list of arguments, e.g. in the middle between two package names which suggests a very different behavior (applying `PUBLIC` only to the following dependency).

I am not sure what a reasonable approach would be. Maybe check the first argument explicitly if it is `PUBLIC`?

Reply:
Yeah, that's right. Now, I'm explicitly checking if the first argument was PUBLIC, and raised a FATAL_ERROR if it was in other place.